frontend/src/components/pagination.tsx

```diff
@@ -0,0 +1,110 @@
+import {
+  ChevronLeft,
+  ChevronRight,
+  ChevronsLeft,
+  ChevronsRight
+} from "lucide-react";
+import React from "react";
+import { Button } from "./ui/button";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue
+} from "./ui/select";
+
+export type PaginationProps = {
+  totalPages: number;
+  currentPage: number;
+  perPage: number;
+  className?: string;
+  onChangePage: (page: number) => void;
+  onChangePerPage: (perPage: number) => void;
+};
+
+const Pagination: React.FC<PaginationProps> = ({
+  totalPages,
+  currentPage,
+  perPage,
+  className = "",
+  onChangePage,
+  onChangePerPage
+}) => {
```

Comment:
```suggestion
export function Pagination({
  totalPages,
  currentPage,
  perPage,
  className = "",
  onChangePage,
  onChangePerPage
}: PaginationProps) => {
```

No `React.FC`, use regular functions